### PR TITLE
Support for RHODS brew configuration

### PR DESCRIPTION
**Pull request by omrirh**
- Adding RHODS brew configuration to `install_addon` flow

Depended on [Support for creating/updating secrets](https://github.com/RedHatQE/openshift-python-utilities/pull/193)

### CODE REVIEW

File: ocm_python_wrapper/cluster.py [hunk 1]
```diff
@@ -388,3 +396,24 @@ def _wait_for_rhmi_resource():
         ResourceEditor(
             patches={rhmi: {"spec": {"useClusterStorage": "false"}}}
         ).update()
+
+    @staticmethod
+    def create_rhods_brew_config():
+        if not os.getenv("PS_DATA"):
```
Comment (myakove) [inline ocm_python_wrapper/cluster.py:402]: Please rename `PS_DATA` use full names

File: ocm_python_wrapper/cluster.py [hunk 2]
```diff
@@ -388,3 +396,24 @@ def _wait_for_rhmi_resource():
...
+            raise KeyError("Pull-secret data must be specified")
```
Comment (myakove) [inline ocm_python_wrapper/cluster.py:403]: `Pull-secret data must be specified")` specified where?

File: ocm_python_wrapper/cluster.py [hunk 3]
```diff
@@ -388,3 +396,24 @@ def _wait_for_rhmi_resource():
...
+            icsp_name="brew-registry",
+            repository_digest_mirrors=[
+                {
+                    "source": "registry.redhat.io/rhods",
+                    "mirrors": ["brew.registry.redhat.io/rhods"],
+                }
+            ],
+        )
+        brew_pull_secret = create_update_pull_secret(
+            new_content_dict=json.loads(os.environ["PS_DATA"]),
```
Comment (myakove) [inline ocm_python_wrapper/cluster.py:415]: `os.environ["PS_DATA"]` save and use it.

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocm_python_wrapper/cluster.py (hunk 1 above)
Comment (omrirh) [reply ocm_python_wrapper/cluster.py]: done

File: ocm_python_wrapper/cluster.py (hunk 2 above)
Comment (omrirh) [reply ocm_python_wrapper/cluster.py]: Updated error message with env var name to pass the pull-secret data

File: ocm_python_wrapper/cluster.py (hunk 3 above)
Comment (omrirh) [reply ocm_python_wrapper/cluster.py]: done

File: ocm_python_wrapper/cluster.py [hunk 4]
```diff
@@ -388,3 +396,24 @@ def _wait_for_rhmi_resource():
...
+            ps_name="pull-secret",  # pragma: allowlist secret
+            ps_namespace="openshift-config",  # pragma: allowlist secret
```
[the remark this answers is not in the document]
Comment (omrirh) [reply ocm_python_wrapper/cluster.py]: It is to satisfy the secret detector linter, it spots the keyword `secret` and fails the ci

File: ocm_python_wrapper/cluster.py [hunk 5]
```diff
@@ -388,3 +396,24 @@ def _wait_for_rhmi_resource():
...
+        )
+        return brew_icsp, brew_pull_secret
```
[the remark this answers is not in the document]
Comment (omrirh) [reply ocm_python_wrapper/cluster.py]: Not needed in this case. removing the returned values

**Review by myakove — CHANGES_REQUESTED** [review]

File: ocm_python_wrapper/cluster.py [hunk 6]
```diff
@@ -388,3 +396,29 @@ def _wait_for_rhmi_resource():
         ResourceEditor(
             patches={rhmi: {"spec": {"useClusterStorage": "false"}}}
         ).update()
+
+    @staticmethod
+    def create_rhods_brew_config():
+        secret_data_name = "PULL_SECRET"  # pragma: allowlist secret
+
+        try:
```
[the remark this answers is not in the document]
Comment (omrirh) [reply ocm_python_wrapper/cluster.py]: done

File: ocm_python_wrapper/cluster.py [hunk 7]
```diff
@@ -388,3 +396,29 @@ def _wait_for_rhmi_resource():
         ResourceEditor(
             patches={rhmi: {"spec": {"useClusterStorage": "false"}}}
         ).update()
+
+    @staticmethod
+    def create_rhods_brew_config():
+        secret_data_name = "PULL_SECRET"  # pragma: allowlist secret
```
[the remark this answers is not in the document]
Comment (omrirh) [reply ocm_python_wrapper/cluster.py]: done

File: ocm_python_wrapper/cluster.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (omrirh) [reply ocm_python_wrapper/cluster.py]: removed

**Review by rnetser — CHANGES_REQUESTED** [review]

**Review by rnetser — CHANGES_REQUESTED** [review]
see comment

File: ocm_python_wrapper/cluster.py [hunk 8]
```diff
@@ -388,3 +396,28 @@ def _wait_for_rhmi_resource():
...
+
+    @staticmethod
+    def create_rhods_brew_config():
+        secret_data_name = "BREW_PULL_SECRET"  # pragma: allowlist secret
+
+        if not os.getenv(secret_data_name):
+            raise KeyError(
+                f"Pull-secret data must be specified in {secret_data_name} environment variable"
+            )
+
```
[the remark this answers is not in the document]
Comment (omrirh) [reply ocm_python_wrapper/cluster.py]: done

**Review by myakove — APPROVED** [review]

File: ocm_python_wrapper/cluster.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (omrirh) [reply ocm_python_wrapper/cluster.py]: @rnetser considering passing the pull secret via a file, should that filename be passed via addon parameters ?

**Review by rnetser — APPROVED** [review]

**Review by rnetser — APPROVED** [review]

**Comment by rnetser** [comment]
@omrirh you need to cherrypick https://github.com/RedHatQE/openshift-python-utilities/pull/193 before this one can be marked as `verified` and merged

**Review by rnetser — CHANGES_REQUESTED** [review]
See comment on cherrypick

**Review by rnetser — APPROVED** [review]

**Pull request merged**